fenics_ice/solver.py

```diff
@@ -492,14 +496,34 @@ def sliding_law(self, alpha, U):
         if sl == 'linear':
             B2 = C
 
-        elif sl == 'budd':
-            N = (1-fl_ex)*(H*rhoi*g + ufl.Min(bed, 0.0)*rhow*g)
+        elif ((sl == 'budd') or (sl == 'corn')):
             U_mag = sqrt(U[0]**2 + U[1]**2 + vel_rp**2)
-            # Need to catch N <= 0.0 here, as it's raised to
-            # 1/3 (forward) and -2/3 (adjoint)
-            N_term = ufl.conditional(N > 0.0, N ** (1.0/3.0), 0)
-            B2 = (1-fl_ex)*(C * N_term * U_mag**(-2.0/3.0))
+            N = (1-fl_ex)*(H*rhoi*g + ufl.Min(bed, 0.0)*rhow*g)
 
+            if sl == 'budd':
+                # Need to catch N <= 0.0 here, as it's raised to
+                # 1/3 (forward) and -2/3 (adjoint)
+                N_term = ufl.conditional(N > 0.0, N ** (1.0/3.0), 0)
+                B2 = (1-fl_ex)*(C * N_term * U_mag**(-2.0/3.0))
+
+            elif sl == 'corn':
+                # see eq 11 of Asay Davis et al, Experimental design for 
+                # three interrelated marine ice sheet and ocean model intercomparison projects ... 
+                # Geosci. Model Dev., 9, 2471–2497
+
+                # Need to catch N <= 0.0 here, as it's raised to
+                # 1/3 (forward) and -2/3 (adjoint)
+                LocalProjectionSolver(H, self.H_DG).solve() 
+                LocalProjectionSolver(U_mag, self.Umag_DG).solve()
```

Comment:
@jrmaddison i do suspect it is choosing a quadrature order = 23 because of a single term, but if i choose a more reasonable quadrature order, this will essentially have the same effect, without greatly complicating the weak form!

Reply:
Perhaps we could avoid over-integration of the simple terms, but admittedly this unlikely to be a serious performance problem. Fixing at a reasonable degree seems fine.